sim/core/spell.go

```diff
@@ -450,10 +476,10 @@ func (spell *Spell) CanCast(sim *Simulation, target *Unit) bool {
 		return false
 	}
 
-	// While casting, no other action is possible
+	// While casting or channeling, no other action is possible
```

Comment:
This is no longer the correct condition to check for channels, so it's intentional for the comment to not mention channels.

Also, channeling doesn't prevent casts, that's what recasts are. So it's important that this function not return false just because the agent is channeling. I ran into this recently while adding the Channel APL action.

Reply:
Channeling still sets a "hardcast" though, so effectively it _does_ prevent other cast actions.

By "recast" I assume you mean cancelling the current "hardcast", then casting again? That's fine, but doesn't affect this comment, because then there's no "hardcast" happening anymore.